sda-commons-server-jackson/src/main/java/org/sda/commons/server/jackson/hal/HalLinkProvider.java

```diff
@@ -0,0 +1,154 @@
+package org.sda.commons.server.jackson.hal;
+
+import io.openapitools.jackson.dataformat.hal.HALLink;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Feature;
+import javax.ws.rs.core.FeatureContext;
+import javax.ws.rs.core.UriBuilder;
+import javax.ws.rs.core.UriInfo;
+import org.glassfish.jersey.uri.internal.JerseyUriBuilder;
+import org.sda.commons.server.jackson.hal.HalLinkInvocationStateUtility.MethodInvocationState;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Helper utility which is registered as singleton instance in the jersey environment via the
+ * {@linkplain org.sdase.commons.server.jackson.JacksonConfigurationBundle}. It provides the
+ * context-based HALLink for a JAX-RS interface using the {@linkplain javax.ws.rs.PathParam} and
+ * {@linkplain javax.ws.rs.QueryParam} annotations.
+ *
+ * <p>Usage:
+ *
+ * <pre>
+ * &#064;Path("")
+ * interface TestApi {
+ *  &#064;Path("/testPath/{testArg}")
+ *  &#064;GET
+ *  String testMethod(@PathParam("testArg") String testArg, @QueryParam("query") query);
+ * }
+ *
+ * // Get the generated HALLink
+ * HALLink HalLink = linkTo(methodOn(TestApi.class).testMethod("ResourceID", "detailed")).asHalLink();
+ * // Get the generated URI
+ * URI Uri = linkTo(methodOn(TestApi.class).testMethod("ResourceID", "detailed")).asURI();
+ * </pre>
+ *
+ * The example would create the following URI: {@code "baseUri/testPath/ResourceID?query=detailed"}
+ */
+public class HalLinkProvider implements Feature {
+
+  private static final Logger LOG = LoggerFactory.getLogger(HalLinkProvider.class);
+  private static HalLinkProvider instance;
+
+  @Context private UriInfo uriInfo;
+
+  private HalLinkProvider() {}
+
+  /**
+   * Creates a {@linkplain HALLink} based on JAX-RS annotated parameters of an interface. This
+   * method requires a second entrypoint with {@linkplain
+   * HalLinkInvocationStateUtility#methodOn(Class)} to process the annotated JAX-RS interface with
+   * the corresponding {@linkplain PathParam} and {@linkplain QueryParam} and the passed arguments
+   * to the proxied method invocation.
+   *
+   * <p>Example:
+   *
+   * <pre>
+   * &#064;Path("")
+   * interface TestApi {
+   *  &#064;Path("/testPath/{testArg}")
+   *  &#064;GET
+   *  String testMethod(@PathParam("testArg") String testArg, @QueryParam("query") query);
+   * }
+   *
+   * // Get the generated HALLink
+   * HALLink halLink = linkTo(methodOn(TestApi.class).testMethod("ResourceID", "detailed")).asHalLink();
+   * // Get the generated URI
+   * URI uri = linkTo(methodOn(TestApi.class).testMethod("ResourceID", "detailed")).asURI();
+   * </pre>
+   *
+   * The example would create the following URI: {@code
+   * "baseUri/testPath/ResourceID?query=detailed"}
+   *
+   * @param invocation the invocation placeholder.
```

Comment:
```suggestion
   * <p>The builder will use the `UriInfo` of the current request context to create absolute URIs. If no request context is available, it will fall back to a simple path without host that does not include any configured root or context path.
   * @param invocation the invocation placeholder.
```

Reply:
Added 👍